lms/services/blackboard_api/client.py

```diff
@@ -63,24 +64,29 @@ def list_files(self, course_id, folder_id=None):
             if not path:
                 break
 
-        # Notify that we've found some files
-        self._request.registry.notify(
-            FilesDiscoveredEvent(
-                request=self._request,
-                values=[
-                    {
-                        "type": "blackboard_file"
-                        if file["type"] == "File"
-                        else "blackboard_folder",
-                        "course_id": course_id,
-                        "lms_id": file["id"],
-                        "name": file["name"],
-                        "size": file["size"],
-                        "parent_lms_id": file["parentId"],
-                    }
-                    for file in results
-                ],
-            )
+        application_instance = self._request.find_service(
+            name="application_instance"
+        ).get_current()
```

Comment:
Not sure if this is right. should the service be injected on the constructor here?

Reply:
I guess we usually inject these into the constructors, yes. Though I don't think there's a huge difference.

If we had a `FileService.upsert()` for `BlackboardAPIClient` to call (as suggested in another comment) then I don't think `BlackboardAPIClient` would need the `application_instance` here and it looks like `FileService` already has the `application_instance` injected